geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/ConnectCommand.java

```diff
@@ -161,17 +161,21 @@ public ResultModel connect(
     }
 
     String gfshVersion = gfsh.getVersion();
+    String gfshGeodeVersion = gfsh.getGeodeVersion();
     String remoteVersion = null;
     try {
-      remoteVersion = invoker.getRemoteVersion();
-      if (versionComponent(remoteVersion, VERSION_MAJOR)
-          .equalsIgnoreCase(versionComponent(gfshVersion, VERSION_MAJOR))
-          && versionComponent(remoteVersion, VERSION_MINOR)
-              .equalsIgnoreCase(versionComponent(gfshVersion, VERSION_MINOR))) {
+      String remoteGeodeVersion = invoker.getRemoteGeodeVersion();
+      if (hasSameMajorMinor(gfshGeodeVersion, remoteGeodeVersion)) {
         return result;
       }
     } catch (Exception e) {
-      gfsh.logInfo("failed to get the the remote version.", e);
+      // we failed to get the remote geode version, now try to get the remote product version for
+      // error message
+      try {
+        remoteVersion = invoker.getRemoteVersion();
```

Comment:
I don't understand why we do this. Since we are now comparing the remote geode version, shouldn't we also report it in the error message?

Reply:
It is possible that, for example, there may be no changes in gfsh or serialization protocol between 1.10 and 1.11, so it would be safe for the 1.10 gfsh to talk to the 1.11 cluster or the other way around.  The error message is only printed if truly incompatible, in which case it makes more sense to show the actual product version, not the serialization version level (despite being named after Geode versions, if there were actually no serialization changes in 1.11, the serialization version would remain 1.10).  Would it be clearer to rename `geodeVersion` to `geodeSerializationVersion`?

Reply:
I changed `geodeVersion` to `geodeSerializationVersion`